lib/collection/src/lib.rs

```diff
@@ -383,9 +394,12 @@ impl Collection {
     }
 
     pub async fn before_drop(&mut self) {
-        for (_, mut shard) in self.shards.drain() {
-            shard.before_drop().await
-        }
+        let futures: FuturesUnordered<_> = self
```

Comment:
nitpick: Depending on the number of shards, this could generates a lot of concurrent work.

In general I prefer to bound the number of in-flight item to not exhaust resources.
Is it possible to do this here? e.g. have at most 3 concurrents `before_drop`.

You can also discard this comment if you think that this not a real issue for a production load :factory:

Reply:
I trust you on this one :+1: